modules/swagger-jaxrs/src/main/java/com/wordnik/swagger/jaxrs/Reader.java

```diff
@@ -70,9 +102,31 @@ public Reader(Swagger swagger, ReaderConfig config) {
     this.config = new DefaultReaderConfig(config);
   }
 
-  public Swagger read(Set<Class<?>> classes) {
+  public synchronized Swagger read(Set<Class<?>> classes) {
+
+    listeners.clear();
+
+    for(Class<?> cls: classes) {
+      if( ReaderListener.class.isAssignableFrom( cls ) && listeners.containsKey( cls )){
```

Comment:
makes sense - the only objection I have is that you need to know at code-time which listeners you want; I was aiming at a solution where listeners can be added later without having to modify the source code (since the main scan method will find any class implementing this interface, those classes could be added later, for example by scanning a 3rd party library that adds additional Swagger-related features).